src/app/shared/filer-by-property/filter-by-property.pipe.ts

```diff
@@ -0,0 +1,24 @@
+import {Pipe, PipeTransform} from '@angular/core';
+
+@Pipe({
+    name: 'filterByProperty',
+})
+export class FilterByPropertyPipe implements PipeTransform {
+    transform<T>(items: T[] | null, propertyName: keyof T, searchPropertyValue: unknown): T[] {
+        if (!items || !propertyName || !searchPropertyValue) {
+            return [];
+        }
+
+        return items.filter(item => {
+            if (Object.prototype.hasOwnProperty.call(item, propertyName)) {
+                const propertyValue = item[propertyName];
+
+                if (typeof propertyValue === typeof searchPropertyValue) {
+                    return propertyValue === searchPropertyValue;
+                }
+            }
+
+            return false;
```

Comment:
Рекомендация: не плохо было бы добавить ранний `return` и тем самым уменьшить вложенность `if`)
```suggestion
            const hasSearchedProperty = Object.prototype.hasOwnProperty.call(item, propertyName);
            
            if (!hasSearchedProperty) {
                return false;
            } 
            
            const propertyValue = item[propertyName];
            const isConsistentValueType = typeof propertyValue === typeof searchPropertyValue;
            
            return isConsistentValueType && propertyValue === searchPropertyValue;
```